app/components/Views/NetworkSelector/RpcSelectionModal/RpcSelectionModal.tsx

```diff
@@ -49,13 +51,51 @@ interface RpcSelectionModalProps {
 const RpcSelectionModal: FC<RpcSelectionModalProps> = ({
   showMultiRpcSelectModal,
   closeRpcModal,
-  onRpcSelect,
   rpcMenuSheetRef,
   networkConfigurations,
   styles,
 }) => {
   const isAllNetwork = useSelector(selectIsAllNetworks);
 
+  const { navigate } = useNavigation();
+
+  const onRpcSelect = useCallback(
+    async (clientId: string, chainId: `0x${string}`) => {
```

Comment:
Can we use Hex from metamask/utils?
```suggestion
    async (clientId: string, chainId: Hex) => {
```